napari/layers/vectors/vectors.py

```diff
@@ -593,83 +598,57 @@ def _view_face_color(self) -> np.ndarray:
 
-        dims_not_displayed = self._slice_input.not_displayed
+        # The new slicing code makes a request from the existing state and
+        # executes the request on the calling thread directly.
+        # For async slicing, the calling thread will not be the main thread.
+        request = self._make_slice_request_internal(
+            self._slice_input, self._slice_indices
+        )
+        response = request()
+        self._update_slice_response(response)
 
-        # We want a numpy array so we can use fancy indexing with the non-displayed
-        # indices, but as dims_indices can (and often/always does) contain slice
-        # objects, the array has dtype=object which is then very slow for the
-        # arithmetic below.
-        # promote slicing plane to array so we can index into it, project as type float
-        not_disp_indices = np.array(dims_indices)[dims_not_displayed].astype(
-            float
+    def _make_slice_request(self, dims) -> _VectorSliceRequest:
+        """Make a Vectors slice request based on the given dims and these data."""
+        slice_input = self._make_slice_input(
+            dims.point, dims.ndisplay, dims.order
         )
-        # get the anchor points (starting positions) of the vector layers in not displayed dims
-        data = self.data[:, 0, dims_not_displayed]
-        # calculate distances from anchor points to the slicing plane
-        distances = abs(data - not_disp_indices)
-        # if we need to include vectors that are out of this slice
-        if self.out_of_slice_display is True:
-            # get the scaled projected vectors
-            projected_lengths = abs(
-                self.data[:, 1, dims_not_displayed] * self.length
-            )
-            # find where the distance to plane is less than the scaled vector
-            matches = np.all(distances <= projected_lengths, axis=1)
-            alpha_match = projected_lengths[matches]
-            alpha_match[alpha_match == 0] = 1
-            alpha_per_dim = (alpha_match - distances[matches]) / alpha_match
-            alpha_per_dim[alpha_match == 0] = 1
-            alpha = np.prod(alpha_per_dim, axis=1).astype(float)
-        else:
-            matches = np.all(distances <= 0.5, axis=1)
-            alpha = 1.0
-        slice_indices = np.where(matches)[0].astype(int)
-        return slice_indices, alpha
+        # TODO: [see Image]
+        #   For the existing sync slicing, slice_indices is passed through
+        # to avoid some performance issues related to the evaluation of the
+        # data-to-world transform and its inverse. Async slicing currently
+        # absorbs these performance issues here, but we can likely improve
+        # things either by caching the world-to-data transform on the layer
+        # or by lazily evaluating it in the slice task itself.
+        slice_indices = slice_input.data_indices(
+            self._data_to_world.inverse, round_index=False
```

Comment:
`round_index=False` is a change of behavior. See my other comment: https://github.com/napari/napari/pull/5918/files#r1225443605 which is realize is wrong now :D